src/pages/task/ActionButtons.tsx

```diff
@@ -0,0 +1,248 @@
+    },
+    onError: (err) => {
+      error(`Error aborting task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [restartTask, { loading: loadingRestartTask }] = useMutation<
+    RestartTaskMutation,
+    RestartTaskMutationVariables
+  >(RESTART_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task scheduled to restart");
+    },
+    onError: (err) => {
+      error(`Error restarting task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [setTaskPriority, { loading: loadingSetPriority }] = useMutation<
+    SetTaskPriorityMutation,
+    SetTaskPriorityMutationVariables
+  >(SET_TASK_PRIORTY, {
+    onCompleted: (data) => {
+      success(`Priority for task updated to ${data.setTaskPriority.priority}`);
+    },
+    onError: (err) => {
+      error(`Error updating priority for task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const disabled =
+    loadingAbortTask ||
+    loadingRestartTask ||
+    loadingSetPriority ||
+    loadingUnscheduleTask ||
+    loadingScheduleTask;
+
+  useEffect(() => {
+    if (disabled) {
+      setIsVisible(false);
+    }
+  }, [disabled]);
+
+  useOnClickOutside(wrapperRef, () => {
+    if (
+      !get(priorityRef, "current.className", "").includes("ant-popover-open")
+    ) {
+      setIsVisible(false);
+    }
+  });
+
+  const rowButtons = [
+    <Button
+      size="small"
+      dataCy="schedule-task"
+      key="schedule"
+      disabled={disabled || !canSchedule}
+      loading={loadingScheduleTask}
+      onClick={scheduleTask}
+    >
+      Schedule
+    </Button>,
+    <Button
+      size="small"
+      dataCy="restart-task"
+      key="restart"
+      disabled={disabled || !canRestart}
+      loading={loadingRestartTask}
+      onClick={restartTask}
+    >
+      Restart
+    </Button>,
+    <Button
+      size="small"
+      dataCy="notify-task"
+      key="notifications"
+      disabled={disabled}
+    >
+      Add Notification
+    </Button>,
+  ];
+
+  const cardItems = [
+    <CardItem
+      disabled={disabled || !canUnschedule}
+      key="unschedule"
+      data-cy="unschedule-task"
+      onClick={() => unscheduleTask()}
+    >
+      <Body>Unschedule</Body>
+    </CardItem>,
+    <CardItem
+      data-cy="abort-task"
+      key="abort"
+      disabled={disabled || !canAbort}
+      onClick={() => abortTask()}
+    >
+      <Body>Abort</Body>
+    </CardItem>,
+    <Popconfirm
+      key="priority"
+      placement="left"
+      title={
+        <>
+          <StyledBody>Set new priority:</StyledBody>
+          <InputNumber
+            size="small"
+            min={0}
+            type="number"
+            max={Number.MAX_SAFE_INTEGER}
+            value={priority}
+            onChange={setPriority}
+          />
+        </>
+      }
+      onConfirm={() =>
+        setTaskPriority({
+          variables: { taskId, priority },
+        })
+      }
+      onCancel={() => setIsVisible(false)}
```

Comment:
these will fail linting. let's turn off `no lambda` because these should be ok. @khelif96 ;)

Reply:
:) i'll open a pr today to update our  eslint rules